server/interceptor/interceptor.go

```diff
@@ -229,129 +221,51 @@ func isLeafOK(leaf *trillian.QueuedLogLeaf) bool {
-		// * Get, Delete and UndeleteTree may read with deleted trees
-		// * ListTrees returns 0 or more trees
-		*trillian.CreateTreeRequest,
-		*trillian.DeleteTreeRequest,
-		*trillian.GetTreeRequest,
-		*trillian.UndeleteTreeRequest,
-		*trillian.ListTreesRequest,
-		// Do not intercept any quota requests, common rules do not apply.
+		// Quota requests
 		*quotapb.CreateConfigRequest,
 		*quotapb.DeleteConfigRequest,
 		*quotapb.GetConfigRequest,
 		*quotapb.ListConfigsRequest,
 		*quotapb.UpdateConfigRequest:
-		return &rpcInfo{doNotIntercept: true}, nil
-	}
-
-	var treeID int64
-	switch req := req.(type) {
-	case treeRequest:
-		treeID = req.GetTree().GetTreeId()
-	case logIDRequest:
-		treeID = req.GetLogId()
-	case mapIDRequest:
-		treeID = req.GetMapId()
-	default:
-		return nil, status.Errorf(codes.Internal, "cannot retrieve treeID from request: %T", req)
-	}
-
-	treeType, readonly, err := getRequestInfo(req)
-	if err != nil {
-		return nil, err
-	}
-
-	kind := quota.Read
-	if !readonly {
-		kind = quota.Write
-	}
-	var specs []quota.Spec
-	switch {
-	case treeType == trillian.TreeType_UNKNOWN_TREE_TYPE:
-		// Don't impose quota on Admin requests.
-		// Sequencing-based replenishment is not tied in any way to Admin, so charging tokens for it
-		// leads to direct leakage.
-		// Admin is meant to be internal and unlikely to be a source of high QPS, in any case.
-	case treeID == 0:
-		specs = []quota.Spec{
-			{Group: quota.User, Kind: kind, User: quotaUser},
-			{Group: quota.Global, Kind: kind},
-		}
-	default:
-		specs = []quota.Spec{
-			{Group: quota.User, Kind: kind, User: quotaUser},
-			{Group: quota.Tree, Kind: kind, TreeID: treeID},
-			{Group: quota.Global, Kind: kind},
-		}
-	}
-
-	tokens := 1
-	switch req := req.(type) {
-	case logLeavesRequest:
-		tokens = len(req.GetLeaves())
-	case mapLeavesRequest:
-		tokens = len(req.GetLeaves())
-	}
-
-	return &rpcInfo{
-		treeID: treeID,
-		opts:   trees.GetOpts{TreeType: treeType, Readonly: readonly},
-		specs:  specs,
-		tokens: tokens,
-	}, nil
-}
-
-func getRequestInfo(req interface{}) (trillian.TreeType, bool, error) {
-	if readonly, ok := getAdminRequestInfo(req); ok {
-		return trillian.TreeType_UNKNOWN_TREE_TYPE, readonly, nil
-	}
-	if readonly, ok := getLogRequestInfo(req); ok {
-		return trillian.TreeType_LOG, readonly, nil
-	}
-	if readonly, ok := getMapRequestInfo(req); ok {
-		return trillian.TreeType_MAP, readonly, nil
-	}
-	return trillian.TreeType_UNKNOWN_TREE_TYPE, false, fmt.Errorf("unmapped request type: %T", req)
-}
-
-func getAdminRequestInfo(req interface{}) (bool, bool) {
-	_, ok := req.(*trillian.UpdateTreeRequest)
-	return false /* readonly */, ok
-}
+		info.getTree = false
+		info.quota = false
+		info.readonly = false // Doesn't really matter as all interceptors are turned off
+
+	// Admin / readonly
+	case *trillian.GetTreeRequest,
+		*trillian.ListTreesRequest:
+		info.getTree = false // Read done by RPC
```

Comment:
Maybe `// Read done within RPC handler` ? (and below)

Reply:
Done.